### PR TITLE
Adding methods in ICC

### PR DESCRIPTION
**Pull request by Juhibhojani**
## Description

Adding methods in ICC and updating documentation

## Checklist

> Before submitting this pull request, kindly verify that the ensuing checkpoints have been reached.

- [x] Have you adhered to the repository's defined coding convention rules?
- [x] Have you updated the 'documentation.md' file with the method/function documentation?
- [x] Have you sent a message along with the result or response?
- [x] Have you used the try-catch technique?
- [x] Has the method/class been added to the documentation (md file)?

## Screenshots

image

## Additional Notes/Comments

> Any additional remarks or suggestions concerning this pull request are welcome.

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

### CODE REVIEW

File: src/scrape_up/icc/icc_rankings.py [hunk 1]
```diff
@@ -93,3 +95,83 @@ def player_ranking(self, type, format):
...
+            for rank, team in enumerate(teams, 1):
+                obj_values = [rank, team.get_text()]
+                resposne_list.append(dict(zip(obj_keys, obj_values)))
+
+            return resposne_list
+        except:
+            return None
+
+    def player_ranking_women(self, type, format):
+        """
```
Comment (nikhil25803) [inline src/scrape_up/icc/icc_rankings.py:133]: Test the method and also change the docstring accordingly I think you just copied from the previous methods.

Also, test the endpoints if they are returning data properly or not in the desired format.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

File: src/scrape_up/icc/icc_rankings.py (hunk 1 above)
Comment (Juhibhojani) [reply src/scrape_up/icc/icc_rankings.py:133]: Data is being returned in proper format and i have updated the docstring

File: src/scrape_up/icc/icc_rankings.py [hunk 2]
```diff
@@ -93,3 +95,83 @@ def player_ranking(self, type, format):
...
+        icc = ICC()
+        icc.player_ranking_women(format="test",type="batting")
+        ```
+        Returns \n
+        ```js
+        [
+            {
+                "rank":1,
+                "team":"Kane Williamson"
+            }
```
Comment (nikhil25803) [inline src/scrape_up/icc/icc_rankings.py:148]: The method isn't working properly, also the docstring is not updated with the response.  I do not think Kane Williamson belongs to women playing nation.

image

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

File: src/scrape_up/icc/icc_rankings.py (hunk 2 above)
Comment (Juhibhojani) [reply src/scrape_up/icc/icc_rankings.py:150]: image
The method is now working and the updated docstring is as this:
[
            {
                "rank":1,
                "team":"Natalie Sciver-Brunt"
            }
            ...
]

**Review by nikhil25803 — APPROVED** [review]

**Pull request merged**